components/abstractions/src/main/java/com/microsoft/kiota/ApiException.java

```diff
@@ -24,7 +24,25 @@ public ApiException(@Nonnull final Throwable cause) {
     }
 
     /** The HTTP status code  for the response*/
-    public int responseStatusCode;
+    private int responseStatusCode;
+
+    /**
+     * Gets the HTTP response status code
+     * @return The response status code from the failed response.
+     */
+    @Nonnull
+    public int getResponseStatusCode() {
+        return responseStatusCode;
+    }
+
+    /**
+     * Sets the HTTP response status code
+     * @param responseStatusCode The response status code to set.
+     */
+    protected void setResponseStatusCode(int responseStatusCode) {
+        Objects.requireNonNull(responseStatusCode);
```

Comment:
```suggestion
```
int's can't be null? or does this method also check for 0 ?

Reply:
no they cannot be null you are right